### PR TITLE
Prefix change for standard tags from labels in kubelet, docker and ECS fargate

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

Standard tags (`service`, `version`, `env`) will be extracted from labels (previously annotations). Using labels is preferred due to the established pattern of use in label selectors.

Prefix changes to distinguish these from autodiscovery annotations and labels:
- In PodSpec labels: `ad.datadoghq.com/` => `tags.datadoghq.com/`
- In container/Dockerfile labels: `com.datadoghq.ad` => `com.datadoghq.tags`

### Motivation

Improve usability and minimize confusion with existing autodiscovery annotations.

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by ahmed-mez — DISMISSED** [review]

**Review by clamoriniere — DISMISSED** [review]
💯

**Review by jdgumz — DISMISSED** [review]
LGTM!

File: pkg/tagger/collectors/kubelet_extract_test.go [hunk 1]
```diff
@@ -533,22 +533,22 @@ func TestParsePods(t *testing.T) {
...
-						"component":         "kube-proxy",
-						"tier":              "node",
-						"k8s-app":           "kubernetes-dashboard",
-						"pod-template-hash": "490794276",
+						"component":                           "kube-proxy",
+						"tier":                                "node",
+						"k8s-app":                             "kubernetes-dashboard",
+						"pod-template-hash":                   "490794276",
+						"tags.datadoghq.com/dd-agent.env":     "production",
+						"tags.datadoghq.com/dd-agent.service": "dd-agent",
```
Comment (djova) [inline pkg/tagger/collectors/kubelet_extract_test.go]: What happens if the pod is annotated with a different service than the container? Is it additive, so container will get both service tags? Would be good to have a test for that as well.

Comment (xornivore) [reply pkg/tagger/collectors/kubelet_extract_test.go]: This should not be the most typical use case, but yes, additive. I am adding a test.

**Review by jdgumz — APPROVED** [review]

**Review by djova — APPROVED** [review]

**Review by CharlyF — APPROVED** [review]

**Pull request merged**